FHIR-Converter/client/src/view/common/diagnostic/used-segments-diagnostic.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License in the project root for license information.
+ */
+
+import * as vscode from 'vscode';
+import localize from '../../../i18n/localize';
+
+export const diagnosticsCollection = vscode.languages.createDiagnosticCollection('vscode-health-fhir-converter.unusedSegments');
+
+export function updateDiagnostics(uri: vscode.Uri, unusedSegments: object[]): void {
+	const diagnostics: vscode.Diagnostic[] = [];
+	const components = getComponentsList(unusedSegments);
+	for (let i = 0; i < components['ranges'].length; i++) {
+		diagnostics.push(createWarningDiagnostic(components['ranges'][i], components['values'][i]));
```

Comment:
Can we wrap range and value in a class and use it like `components[i].range` and `component[i].value`?

Reply:
Thanks, updated it.